File: src/main/java/org/strassburger/lifestealz/commands/EliminateCommand.java
package org.strassburger.lifestealz.commands;

import net.kyori.adventure.text.Component;
import org.bukkit.Bukkit;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.command.TabCompleter;
import org.bukkit.entity.Player;
import org.bukkit.event.player.PlayerKickEvent;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.strassburger.lifestealz.LifeStealZ;
import org.strassburger.lifestealz.util.MessageUtils;
import org.strassburger.lifestealz.util.WebHookManager;
import org.strassburger.lifestealz.storage.PlayerData;

import java.util.List;

public final class EliminateCommand implements CommandExecutor, TabCompleter {
    private final LifeStealZ plugin;

    public EliminateCommand(LifeStealZ plugin) {
        this.plugin = plugin;
    }

    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, String[] args) {
        String targetPlayerName = (args != null && args.length > 0) ? args[0] : null;
        if (targetPlayerName == null) {
            throwUsageError(sender);
            return false;
        }

        Player targetPlayer = Bukkit.getPlayer(targetPlayerName);

        if (targetPlayer == null) {
            throwUsageError(sender);
            return false;
        }

        eliminatePlayer(sender, targetPlayer);
        return true;
    }


    private void eliminatePlayer(CommandSender sender, Player targetPlayer) {
        PlayerData playerData = plugin.getStorage().load(targetPlayer.getUniqueId());
        playerData.setMaxHealth(0.0);
        plugin.getStorage().save(playerData);

        dropPlayerInventory(targetPlayer);

        Component kickmsg = MessageUtils.getAndFormatMsg(false, "eliminatedjoin", "&cYou don't have any hearts left!");
        targetPlayer.kick(kickmsg, PlayerKickEvent.Cause.BANNED);

        sender.sendMessage(MessageUtils.getAndFormatMsg(true, "eliminateSuc", "&7You successfully eliminated &c%player%&7!",
                new MessageUtils.Replaceable("%player%", targetPlayer.getName())));

        if (plugin.getConfig().getBoolean("announceElimination")) {
            Component elimAnnouncementMsg = MessageUtils.getAndFormatMsg(true, "eliminateionAnnouncementNature", "&c%player% &7has been eliminated!",
                    new MessageUtils.Replaceable("%player%", targetPlayer.getName()));
            Bukkit.broadcast(elimAnnouncementMsg);
        }

        plugin.getEliminatedPlayersCache().addEliminatedPlayer(targetPlayer.getName());

        plugin.getWebHookManager().sendWebhookMessage(WebHookManager.WebHookType.ELIMINATION, targetPlayer.getName(), sender.getName());
    }

    private void dropPlayerInventory(Player targetPlayer) {
        for (ItemStack item : targetPlayer.getInventory().getContents()) {
            if (item != null) targetPlayer.getWorld().dropItem(targetPlayer.getLocation(), item);
        }
        targetPlayer.getInventory().clear();
    }

    private void throwUsageError(CommandSender sender) {
        Component usageMessage = MessageUtils.getAndFormatMsg(false, "usageError", "&cUsage: %usage%", new MessageUtils.Replaceable("%usage%", "/eliminate <player>"));
        sender.sendMessage(usageMessage);
    }

    @Override
    public @Nullable List<String> onTabComplete(@NotNull CommandSender sender, @NotNull Command command, @NotNull String alias, String[] args) {
        return null;
    }
}

File: src/main/java/org/strassburger/lifestealz/commands/EliminateCommand.kt
package org.strassburger.lifestealz.commands

import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.event.player.PlayerKickEvent
import org.strassburger.lifestealz.Lifestealz
import org.strassburger.lifestealz.util.ManagePlayerdata
import org.strassburger.lifestealz.util.Replaceable

class EliminateCommand : CommandExecutor {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>?): Boolean {
        val targetPlayerName = args?.getOrNull(0)

        if (targetPlayerName == null) {
            throwUsageError(sender)
            return false
        }

        val targetPlayer = Bukkit.getPlayer(targetPlayerName)

        if (targetPlayer == null) {
            throwUsageError(sender)
            return false
        }

        ManagePlayerdata().manageHearts(player = targetPlayer, amount = 0.0, direction = "set")

        for (item in targetPlayer.inventory.contents) {
            if (item != null) targetPlayer.world.dropItem(targetPlayer.location, item)
        }

        targetPlayer.inventory.clear()
        val kickmsg = Lifestealz.getAndFormatMsg(false, "messages.eliminatedjoin", "&cYou don't have any hearts left!")
        targetPlayer.kick(kickmsg, PlayerKickEvent.Cause.BANNED)

        sender.sendMessage(Lifestealz.getAndFormatMsg(true, "messages.eliminateSuc", "&7You successfully eliminated &c%player%&7!", Replaceable("%player%", targetPlayer.name)))

        if (Lifestealz.instance.config.getBoolean("announceElimination")) {
            val elimAannouncementMsg = Lifestealz.getAndFormatMsg(true, "messages.eliminateionAnnouncementNature", "&c%player% &7has been eliminated!", Replaceable("%player%", targetPlayer.name))
            Bukkit.broadcast(elimAannouncementMsg)
        }
        return false
    }

    private fun throwUsageError(sender: CommandSender) {
        val usageMessage = Lifestealz.getAndFormatMsg(false, "messages.usageError", "&cUsage: %usage%", Replaceable("%usage%", "/eliminate <player>"))
        sender.sendMessage(usageMessage)
    }
}